subprojects/core/src/test/groovy/org/gradle/api/internal/attributes/DefaultMutableAttributeContainerTest.groovy

```diff
@@ -61,6 +82,11 @@ class DefaultMutableAttributeContainerTest extends Specification {
 
         expect:
         container.asImmutable().keySet() == [secondAttribute, firstAttribute] as Set
+
+        and:
+        def events = outputEventListener.events.findAll { it.logLevel == LogLevel.WARN }
+        events.size() == 1
+        events[0].message.startsWith("Querying the contents of an attribute container while realizing attributes of the container. This behavior has been deprecated. This will fail with an error in Gradle 9.0")
```

Comment:
```suggestion
        events[0].message.startsWith("Querying the contents of an attribute container while realizing attribute "first" within that container is not allowed. This behavior has been deprecated. This will fail with an error in Gradle 9.0")
```

Reply:
This is not always triggered due to querying a single attribute. 

Calling `toString`, calling `equals`, etc, can cause this deprecation

Reply:
Wouldn't those calls result in realizing a particular first attribute that makes it fail? It is probably not useful to display the attribute's name in those cases, though.